resources/page/permalinks.go

```diff
@@ -263,3 +272,80 @@ func (l PermalinkExpander) pageToPermalinkSection(p Page, _ string) (string, err
 func (l PermalinkExpander) pageToPermalinkSections(p Page, _ string) (string, error) {
 	return p.CurrentSection().SectionsPath(), nil
 }
+
+var (
+	nilSliceFunc = func(s []string) []string {
+		return nil
+	}
+	allSliceFunc = func(s []string) []string {
+		return s
+	}
+)
+
+func (l PermalinkExpander) toSliceFunc(cut string) func(s []string) []string {
+	cut = strings.ToLower(strings.TrimSpace(cut))
+	if cut == "" {
+		return allSliceFunc
+	}
+
+	if len(cut) < 3 || (cut[0] != '[' || cut[len(cut)-1] != ']') {
+		return nilSliceFunc
+	}
+
+	toNFunc := func(s string, low bool) func(ss []string) int {
```

Comment:
It doesn't look like `toNFunc` uses any resources outside of its local params.  I'd recommend moving this outside of this function to prevent recreating the same func on each call.

Reply:
`toSliceFunc` is called very few times (normally 0 and 1) per build and this is cheap enough that I prefer that have it defined close to where it is used.